core/src/main/java/org/infinispan/partitionhandling/impl/PartitionHandlingManagerImpl.java

```diff
@@ -186,16 +188,18 @@ public void onTopologyUpdate(CacheTopology cacheTopology) {
          if (log.isDebugEnabled()) {
             log.debugf("On stable topology update. Pending txs: %d", partialTransactions.size());
          }
+         AtomicBoolean logException = new AtomicBoolean(true);
          for (TransactionInfo transactionInfo : partialTransactions.values()) {
             if (log.isTraceEnabled()) {
                log.tracef("Completing transaction %s", transactionInfo.getGlobalTransaction());
             }
-            completeTransaction(transactionInfo, cacheTopology);
+            completeTransaction(transactionInfo, cacheTopology, logException);
```

Comment:
I'd rather stop trying to complete other transactions after the first `CacheNotFoundResponse` and log the topology change message here.

Reply:
the method is async and it does not wait for the transactions to complete/fail.

Reply:
Sorry, I missed that when I first proposed that you only log once "Another partition or topology changed for while completing partial completed transaction" once. I'm now convinced you should never log it at all :)

When the topology changes during a regular commit we don't log any warning, we just wait for the next topology and retry. Commits of partial transactions aren't any different IMO.

However, now that I know the transactions are completed asynchronously, it is more obvious that there is no DEBUG-level log to say "Finished completing partial transactions", or even "Finished sending commit commands" for all partial transactions. That would be more interesting to me, because it would help debug weird situations like 2 stable topologies in a quick succession that trigger 2 completions for the same transactions in parallel.

Reply:
added debug message after the loop :)